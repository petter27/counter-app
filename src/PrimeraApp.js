import React from 'react';
import PropTypes from 'prop-types';


const PrimeraApp = ({saludo, subtitulo}) => {

  return ( 
    <>
      <h1>{ saludo }</h1> 
      <p>{ subtitulo }</p>
    </>
  );
}
// Esto es una forma de documentar y solicitar a otros desarrolladores que envien los props cuando utilicen el componente 
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
};

// esto lo que hace es crear los props o asignarles un valor por defecto cuando no se envia nada.
PrimeraApp.defaultProps = {
  saludo: '',
  subtitulo : 'Este es un proptype por default'
}
 
export default PrimeraApp;