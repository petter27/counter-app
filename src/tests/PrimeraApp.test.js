import React from 'react';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en componente Primera App', ()=>{
/*
  test('El saludo debe ser "Hola soy Goku"', ()=>{

    const saludo = 'Hola soy goku';

    // Producto renderizado del componente que se quiere mostrar.
    const { getByText } = render( <PrimeraApp saludo={ saludo } /> );

    // eslint-disable-next-line testing-library/prefer-screen-queries
    expect( getByText(saludo) ).toBeInTheDocument();
  }); */
  test('Debe de mostrar <PrimeraApp/> Correctamente', ()=>{

    const saludo = 'Hola soy Goku';

    const wrapper = shallow(<PrimeraApp saludo={ saludo } />);

    expect( wrapper ).toMatchSnapshot();

  });
});