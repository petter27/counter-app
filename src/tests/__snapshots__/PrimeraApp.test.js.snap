// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente Primera App Debe de mostrar <PrimeraApp/> Correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Goku
  </h1>
  <p>
    Este es un proptype por default
  </p>
</Fragment>
`;
